src/modules/navigator/navigator_main.cpp

```diff
@@ -370,6 +370,83 @@ void Navigator::run()
 
 				// CMD_DO_REPOSITION is acknowledged by commander
 
+			} else if (cmd.command == vehicle_command_s::VEHICLE_CMD_DO_CHANGE_ALTITUDE
+				   && _vstatus.arming_state == vehicle_status_s::ARMING_STATE_ARMED) {
```

Comment:
Or should commander reject when disarmed?

Reply:
Sounds cleaner to me too yes. @bkueng agree?

Reply:
Yes

Reply:
I actually actively decided against it originally, see https://github.com/PX4/PX4-Autopilot/pull/19776

Reply:
I'm in favor of not changing it withing the context of this PR

Reply:
Unless I'm missing some use-case I would just reject it while disarmed.

Reply:
Yes let's do that separately.